YOLayout/UIView+YOLayout.m

```diff
@@ -0,0 +1,119 @@
+//
+//  UIView+YOLayout.m
+//  YOLayoutExample
+//
+//  Created by Lucas Yan on 8/1/16.
+//  Copyright © 2016 YOLayout. All rights reserved.
+//
+
+#import "UIView+YOLayout.h"
+#import <objc/runtime.h>
+#import "YOLayout.h"
+#import "YOCGUtils.h"
+
+static char kAssociatedLayoutKey;
+
+@implementation UIView (YOLayout)
+
++ (void)useYOLayout {
+  [self exchangeImplementation:@selector(initWithFrame:) withImplementation:@selector(_initWithFrame:)];
+  [self exchangeImplementation:@selector(initWithCoder:) withImplementation:@selector(_initWithCoder:)];
+  [self exchangeImplementation:@selector(setFrame:) withImplementation:@selector(_setFrame:)];
+  [self exchangeImplementation:@selector(layoutSubviews) withImplementation:@selector(_layoutSubviews)];
+  [self exchangeImplementation:@selector(sizeThatFits:) withImplementation:@selector(_sizeThatFits:)];
+  [self exchangeImplementation:@selector(setNeedsLayout) withImplementation:@selector(_setNeedsLayout)];
+}
+
++ (void)exchangeImplementation:(SEL)originalImplementation withImplementation:(SEL)replacementImplementation {
+  Method originalMethod = class_getInstanceMethod(self, originalImplementation);
+  Method swizzledMethod = class_getInstanceMethod(self, replacementImplementation);
+  
+  BOOL didAddMethod =
+  class_addMethod(self,
+      originalImplementation,
+      method_getImplementation(swizzledMethod),
+      method_getTypeEncoding(swizzledMethod));
+  
+  if (didAddMethod) {
+    class_replaceMethod(self,
+      replacementImplementation,
+      method_getImplementation(originalMethod),
+      method_getTypeEncoding(originalMethod));
+  } else {
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+  }
+}
+
+- (void)viewInit {
+  
+}
+
+- (CGSize)layoutWithLayout:(YOLayout *)layout size:(CGSize)size {
```

Comment:
Really don't think we need this and `setupDefaultLayout` to be a part of the category. A UIView class could always add this method if it wanted. What's the advantage of having this as part of the category